imagegw/test/imagemngr_test.py

```diff
@@ -892,6 +893,48 @@ def test_pull_acl(self):
         self.images.drop()
         self.stop_worker()
 
+    def test_import(self):
+        """
+        Basic import test including an induced pull failure.
+        """
+        # Use defaults for format, arch, os, ostcount, replication
+        pr = {
+            'system': self.system,
+            'itype': self.itype,
+            'tag': self.tag,
+            'remotetype': 'dockerv2',
+            'filepath': '/images/test/test.squashfs',
+            'format': 'squashfs',
+            'userACL': [],
+            'groupAcl': []
+        }
+        # Do the pull
+        self.start_worker()
+        session = self.m.new_session(self.auth, self.system)
+        rec = self.m.mngrimport(session, pr, testmode=1)  # ,delay=False)
+        assert rec is not None
+        id = rec['_id']
+        # Confirm record
+        q = {'system': self.system, 'itype': self.itype, 'pulltag': self.tag}
+        mrec = self.images.find_one(q)
+        assert '_id' in mrec
+        # Track through transistions
+        state = self.time_wait(id)
+        #Debug
+        assert state == 'READY'
+        imagerec = self.m.lookup(session, pr)
+        assert 'ENTRY' in imagerec
+        assert 'ENV' in imagerec
+        # Cause a failure
+        self.images.drop()
+        rec = self.m.pull(session, pr, testmode=2)
+        time.sleep(10)
+        assert rec is not None
+        id = rec['_id']
+        state = self.m.get_state(id)
+        assert state == 'FAILURE'
+        self.stop_worker()
```

Comment:
I think we need to add another test that tries to do a real import.  I think this would help get the coverage back up.

Reply:
Yes, I think this sounds good. We need to talk about how to get the fasthash.py into the test system.